### ISSUE TITLE
Life Q35 Support

### ISSUE THREAD

**Issue by gmallios** [gmallios/SoundcoreManager#8]
Current status of Life Q35/A3027 support
- [x] Connection
- [x] Fetch device status 
- [ ] ANC
   - [ ] Remove/Don't use non-supported profiles
   - [ ] Check if the Transparency Option uses the same command as A3951
- [x] EQ
- [ ] UI 
    - [x] OverviewCard
    - [ ] ANCModeCard - Remove Custom option and Transparent sub-options
- [ ] Response Checksum Validation - Need to figure out propper response length
    - [ ] DeviceStatus Reponse Length
    - [ ] DeviceInfo Response Length

**Comment by gmallios** [comment]
Test the artifact in this [workflow](https://github.com/gmallios/SoundcoreManager/actions/runs/3966656917)

**Comment by TETYYS** [comment]
Still the same unfortunately. Is there a way I could provide a debugging output or something similar?
![VkG7DpYU2j](https://user-images.githubusercontent.com/3091645/213686829-5c73fd9f-95ac-4cbe-b84a-19c0aa7dbc43.png)
![Soundcore_Manager_rYB8gQENgm](https://user-images.githubusercontent.com/3091645/213686847-d5b15926-cc36-4554-bda8-27d7521333a7.png)

**Comment by gmallios** [comment]
> Still the same unfortunately. Is there a way I could provide a debugging output or something similar? ![VkG7DpYU2j](https://user-images.githubusercontent.com/3091645/213686829-5c73fd9f-95ac-4cbe-b84a-19c0aa7dbc43.png) ![Soundcore_Manager_rYB8gQENgm](https://user-images.githubusercontent.com/3091645/213686847-d5b15926-cc36-4554-bda8-27d7521333a7.png)


An updated build is available [here](https://github.com/gmallios/SoundcoreManager/actions/runs/3968402098). Thanks for testing and for your time!

**Comment by TETYYS** [comment]
![msedgewebview2_AAch2x8DBr](https://user-images.githubusercontent.com/3091645/213725471-85d38c42-73f2-4246-91b7-42fab58c998a.png)

**Comment by gmallios** [comment]
I've disabled checksum validation try this [one](https://github.com/gmallios/SoundcoreManager/actions/runs/3968650528) out

**Comment by TETYYS** [comment]
- While mobile app is opened, desktop app can't connect, when you press on `Connect, it spins for a couple of ms and throws you back to device selection. But I guess this is kinda expected, although if there is a specific error that device reports it could be good to display to user. Same is true vice-versa - phone app just hangs saying "Connecting..." while desktop app is open
 - Three dots at top right are non-pressable (might be a general issue)
 - ⬅at top left is non-pressable (might be a general issue)
 - Headphones image is wrong, I guess just not updated
 - AFAIK Q35 Life only has one battery so display of two individual batteries is not right, phone app displays only one battery indicator
 - Q35 Life in transparency mode doesn't have selection of any additional modes. That is, it can be only transparent and thats it, no `Fully transparent` or `Vocal mode`. The buttons doesn't do anything
 - Q35 Life does not have `Custom` active ANC option, so it doesn't do anything
 - EQ doesn't work (works in phone app)

Very nice work, I mainly searched for a desktop app that allows me to switch between ANC modes and that fully works

**Comment by gmallios** [comment]
@TETYYS Thanks a lot for your comments and for your time testing out my app, I will continue towards fully supporting the A3027 model. Regarding your last comment:
  1. This is expected as this is a Bluetooth/Device limitation (1 open connection). An alert message will be added to inform the user!
  2. This is a general issue as the TopBar is in-progress
  3. Same as the above
  4. Life Q35-Specific UI is not yet implemented this is the next step
  5. Okay, I did not know that. I though most of the devices which supported ANC had the same modes. I will fix that.
  6. Same as the above
  7. This will require some time and testing but it's possible to implement.

**Comment by gmallios** [comment]
@TETYYS Does the displayed EQ wave in the desktop app match the EQ in the Mobile App?

**Comment by TETYYS** [comment]
Yes, there are some inaccuracies in how the estimated graph is rendered, but the points that are should be controllable match

**Comment by gmallios** [comment]
@TETYYS Could you try the [latest](https://github.com/gmallios/SoundcoreManager/actions/runs/3971425397) build to see if the EQ works?

**Comment by TETYYS** [comment]
It doesn't unfortunately, I see the headphone & battery display is fixed though 👍

**Comment by gmallios** [comment]
Sorry I forgot to change the command😰. Please try [this](https://github.com/gmallios/SoundcoreManager/actions/runs/3975643369) one.

**Comment by TETYYS** [comment]
Same result unfortunately

**Comment by gmallios** [comment]
Just to be clear, you can see the EQ and change values but those changes are not applied?

**Comment by TETYYS** [comment]
Yes, the UI changes, but I hear no difference and if I close and open the application I see that no changes were applied, it's back to flat line EQ I had previously.

**Comment by gmallios** [comment]
If you go to the mobile app change the EQ and then connect with the desktop app are the changes reflected?

**Comment by TETYYS** [comment]
Yes. It feels like the desktop app only reads and displays the EQ state one time when it starts, nothing else.

**Comment by gmallios** [comment]
The desktop app should continuously read the EQ, so I don't think thats the problem. Sending the updated EQ is.
I think I'm close to a new build in which EQ hopefully works. I will keep you updated.

**Comment by gmallios** [comment]
I have re-implemented the set_eq method for A3027 after digging through the Soundcore App. You can find the build [here ](https://github.com/gmallios/SoundcoreManager/actions/runs/3980190106). Let me know if it works.

**Comment by TETYYS** [comment]
EQ is fully working 👍

One more thing I noticed is that when audio is playing, connecting through desktop app makes the audio crackle a bit, which doesn't happen on mobile app

**Comment by gmallios** [comment]
Does it constantly do that while connected or only upon the first connection?

**Comment by TETYYS** [comment]
Only upon launching the app, before pressing `Connect`

**Comment by gmallios** [comment]
That's weird. Before connecting the app cannot/doesn't send anything to the device, it only initiates a bluetooth scan. It may be something specific to the bluetooth adaptor that you have.

**Comment by TETYYS** [comment]
That could be, on the desktop I use an external cheap bluetooth dongle. I'll test the app on a laptop if I get a chance.

**Comment by gmallios** [comment]
Perfect, feel free to open an issue if the problem persists.

**Comment by biozob** [comment]
![image](https://user-images.githubusercontent.com/34273696/223396976-7295306b-60eb-4bf0-b02b-66412cc99b21.png)

![image](https://user-images.githubusercontent.com/34273696/223396541-efae9077-e5bc-4433-928a-c760661eba11.png)

I just downloaded the app (SoundcoreManager-windows-latest-59735ef4.exe) and tested it on my Life Q35 and after clickining on connect this window keeps loading for infinite time.

**Comment by gmallios** [comment]
> ![image](https://user-images.githubusercontent.com/34273696/223396976-7295306b-60eb-4bf0-b02b-66412cc99b21.png)
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/34273696/223396541-efae9077-e5bc-4433-928a-c760661eba11.png)
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/34273696/223398439-5f9e2c25-8415-4feb-9f53-266f2b561e0b.png)
> 
> 
> 
> 
> 
> I just downloaded the app (SoundcoreManager-windows-latest-59735ef4.exe) and tested it on my Life Q35 and after clickining on connect this window keeps loading for infinite time.

Will look into this today. Must be a mistake on my end 😕

**Comment by gmallios** [comment]
I had enabled Response Verification for this device which doesn't seem to work. I've pushed a new build and it should work as before now.

**Comment by PiotrSzperka** [comment]
Hi there, I just downloaded latest windows (SoundcoreManager-windows-latest-2a6ea1d0.exe) and there is a similar issue as above:
![obraz](https://user-images.githubusercontent.com/62660295/229468197-1dde8e8c-233a-4b34-9a63-01e1dd6c89de.png)

I can provide you any additional details if you need some.

**Comment by igorbmaciel** [comment]
The same erros to me

**Comment by gmallios** [comment]
> The same errors to me, I'm using the latest version [e7a3cfd](https://github.com/gmallios/SoundcoreManager/commit/e7a3cfd7f54f89c3ce28661868179a40196d3fcc)

I've pushed a new build with better error messages to see what is going on. In 10-20mins it will be ready to download using the "Windows Latest" button

**Comment by igorbmaciel** [comment]
> > The same errors to me, I'm using the latest version [e7a3cfd](https://github.com/gmallios/SoundcoreManager/commit/e7a3cfd7f54f89c3ce28661868179a40196d3fcc)
> 
> I've pushed a new build with better error messages to see what is going on. In 10-20mins it will be ready to download using the "Windows Latest" button

Ok thanks

**Comment by gmallios** [comment]
I had wrong response length checks and they were not triggering with the old windows API. I have updated them and it should work now, wait for the new build.

**Comment by igorbmaciel** [comment]
> I had wrong response length checks and they were not triggering with the old windows API. I have updated them and it should work now, wait for the new build.

Now it's working, thanks 😊
